### PR TITLE
Add linter + typescript + git precommit hook to the project

### LINKED ISSUE #2: Front-end setup
Add packages that are relevant to the project
Typescript, prettier, ESLint config

### PR DESCRIPTION
**Pull request by willyyhuang**
- Added relevant packages to package manager
- TypeScript configuration
- Prettier and ESLint configuration
- Git pre-commit hook

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
     "private": true,
     "scripts": {
-        "dev": "npm run development",
-        "development": "mix",
+        "dev": "npm run development -- --watch",
+        "development": "node_modules/cross-env/src/bin/cross-env.js NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --config=node_modules/laravel-mix/setup/webpack.config.js",
```
Comment (m-triassi) [inline package.json:5]: I believe doing the development script like this is side stepping mix? I think we need to change the package manager in the mix file to use Yarn instead,  see https://github.com/JeffreyWay/laravel-mix/pull/2642 I think they added yarn support officially in Mix 6.X

File: .env.example [hunk 2]
```diff
@@ -1,49 +0,0 @@
...
-AWS_DEFAULT_REGION=us-east-1
-AWS_BUCKET=
-
-PUSHER_APP_ID=
-PUSHER_APP_KEY=
-PUSHER_APP_SECRET=
-PUSHER_APP_CLUSTER=mt1
-
-MIX_PUSHER_APP_KEY="${PUSHER_APP_KEY}"
-MIX_PUSHER_APP_CLUSTER="${PUSHER_APP_CLUSTER}"
```
Comment (m-triassi) [inline .env.example]: This file needs to not be deleted, it serves as the template for anyone installing the project

**Review by m-triassi — CHANGES_REQUESTED** [review]
Over all, looks good, i think we need to just make sure that Yarn is integrated properly and restoring `.env.example` file. 

Also what do you think about `.gitignore`ing the lock files for yarn? is that back practice? i don't remember

File: .env.example (hunk 2 above)
Comment (willyyhuang) [reply .env.example]: sorry, i forgot about it

**Comment by willyyhuang** [comment]
> Over all, looks good, i think we need to just make sure that Yarn is integrated properly and restoring `.env.example` file.
> 
> Also what do you think about `.gitignore`ing the lock files for yarn? is that back practice? i don't remember

`yarn.lock` serves as package-lock for yarn, so it should not be git ignored

File: package.json (hunk 1 above)
Comment (willyyhuang) [reply package.json:5]: done

**Review by m-triassi — APPROVED** [review]
Alright everything looks in order here; Good work, Approved!

_bruh bruh bruh_ 😂

**Pull request merged**